### DISCUSSION TITLE
Discussion: Plugin /frontendlogic - why is it a plugin?

### DISCUSSION THREAD

**Discussion by tone2tone** [CONTENIDO/CONTENIDO#492] [General]
The "old" plugin named /frontendlogic is not a standard plugin. It holds nothing but a php-file with a class named frontendlogic_category that extends the class FrontendLogic with three functions. Those functions are essential to make the rights management configuration in Administration/Frontend/<group>/rights  work.
Strangely enough, the basic class FrontendLogic, hosted in /classes/class.frontend.logic.php is nothing more than an abstract class with the same three functions that expects its "children" to fill them with life. As far as I can see, FrontendLogic servers no other child but just frontendlogic_category... 
Why then do we keep this old plugin/class relation instead of melting everything in one class and give up the old plugin? Does anyone have an explanation for that?

**Comment by dceonline** [comment]
Well, how do i explain with a few words. Frontendlogic was ment as plugin system like frontendusers. One can enhance these logic by putting feature folders into the plugin dir. If u look at the class frontendlogic_category you will see, that it extends the base class FrontendLogic in the contenido/classes folder.
The folder is included in functions.includePluginConf.php, same as done with the folder frontendusers. I don' t know exactly what was the idea behind this, because its real old stuff and there are no real documentation. In doc folder of 4.6 and/or 4.8 there are a real small file about it.

I think we have to decide what way we wanna go with that, before we do anything with this old stuff. Let it go for the moment, since we have enough man power and a concept for CONTENIDO where to go in the future.
